### ISSUE TITLE
[object Object] after storing object after null in useLocalStorage

### ISSUE THREAD

**Issue by xorik** [vueuse/vueuse#576]
```js
import { useLocalStorage } from '@vueuse/core'

const data = useLocalStorage('test', null)
data.value = {hello: 'world'}
```

[Demo](https://play.vueuse.org/#N4IgDghgxg1hDmBTAziAXAbVAOwgW0XRADcBXRAWgBNE8BLEAGhGQHtSAnKQtEU7MDHhMQNZFA50wAFzqtsRAGrkABABFadFQAoAFhAA2AMxV1sKgGIdE2KLoCUp5CogqaxRAdZgz8FaVkDOmkATxcDLwB3ZxD2FWlWFUjJaUQVAFVsOg8OZEMVZTSAGToAIw4ISRQVI1YOAtUAJhUAMhUAZhFOAyJdaWkwZDQAemH+QXgAOihWPGGyShp6YaDS4bMaAA9JlDxJgCtUAF9GHHweEAABBdJkRGHkfWsqETZObiJxoRExCSlZeREADKT0QVAaiHSd38gWCdBQky6HB6vD6AyGoy+Uxmc2u5Fu90elTBlwArJMAAyTACM62wWx2yD2hxAJzOBCIeMQBOGM2sr3YXAuWJ+KD+MjkCl4AGFWBFEFAAeZWCYUHdsLJ8oUVLK8GBWMg4fIMrDZCgkSiQGjBiMxgIhNNZsMuTy+YgyZSaXSGbsDscALrMIx0AzmtCgACCYDAkwW6FAqT1BggqSIr3F0iIdD1dWkKmA-juRVYUEMQISFSQKiONQ4sxUAHIXXdeXVEA2ADrYLszbDIPNUFOuAC8hcQxdLBnLdQQiG0DdS-YbjBU2FIEXsXcH0ggscMqlHwF0ni8aEbkTqBioDaOrKORyAA)

![Screen Shot 2021-06-11 at 08 04 09](https://user-images.githubusercontent.com/347044/121638652-9be9fc80-ca8b-11eb-9efd-a5f4498ace50.png)

If I replace `null` with `{}` it will work as expected

**Comment by jeffrey-dot-li** [comment]
@mxmvshnvsk This issue is caused because null is typed as 'any', which is basically 'string', by default when sent to local storage. The only way I can think of to fix this is to store null as an object by default. In theory when parsing unknown values you can default to assuming it is an object with `JSON.parse`:
```ts
  any: {
    read: (v: any) => (v != null) ? JSON.parse(v) : null,
    write: (v: any) => String(v),
  },
```
but then the code would throw an error if it ran into anything unexpected.

TLDR; It would be *possible* to fix this issue, but I'm not sure if it would comply with best practice and I think the better solution would be to warn against using `null` as the default value in the documentation?

**Comment by mxmvshnvsk** [comment]
@jeffrey-dot-li I tried to solve this problem, please see my [PR](https://github.com/vueuse/vueuse/pull/614).

**Comment by antfu** [comment]
https://github.com/vueuse/vueuse/pull/614#issuecomment-875450160
